gui/src/ts/components/benchmark/BenchmarkAnswerPage.tsx

```diff
@@ -44,86 +48,116 @@ export const BenchmarkAnswerPage: React.FunctionComponent = () => {
     useParams<BenchmarkAnswerRouteParams>(),
     decodeURIComponent
   );
-  const {submissionId, questionId} = routeParams;
 
   const {data, error, loading} = useQuery<
     BenchmarkAnswerPageQuery,
     BenchmarkAnswerPageQueryVariables
-  >(BenchmarkAnswerPageQueryDocument, {variables: routeParams});
-
-  const benchmark = data?.benchmarkById;
-  const dataset = benchmark?.datasetById;
-  const question = dataset?.questionById;
-  const submission = dataset?.submissionById;
-  const answer = submission?.answerByQuestionId;
+  >(BenchmarkAnswerPageQueryDocument, {
+    variables: routeParams,
+  });
 
   return (
-    <BenchmarkFrame
-      routeParams={routeParams}
-      data={data}
-      error={error}
-      loading={loading}
-    >
-      <Grid container direction="column">
-        {/* Show question and answer choices*/}
-        <Grid item container>
-          <Grid item md={6} container direction="column" justify="center">
-            <Grid item>
-              <Typography variant="h4">{question?.text}</Typography>
-            </Grid>
-          </Grid>
-          <Grid item md={6} container direction="column" spacing={3}>
-            {question?.choices.map((choice) => (
-              <Grid item key={choice.label}>
-                <QuestionAnswerChoiceCard choice={choice} />
-              </Grid>
-            ))}
-          </Grid>
-        </Grid>
+    <BenchmarkFrame data={data} error={error} loading={loading}>
+      {({data}) => {
+        const benchmark = data.benchmarkById;
+        const dataset = benchmark?.datasetById;
+        const question = dataset?.questionById;
+        const submission = dataset?.submissionById;
+        const answer = submission?.answerByQuestionId;
 
-        {/* Extra spacing hack */}
-        <Grid item>
-          <br />
-          <br />
-        </Grid>
+        const {benchmarkId, datasetId, submissionId, questionId} = routeParams;
+
+        if (!benchmark) {
+          return <NotFound label={benchmarkId} />;
+        }
+
+        if (!dataset) {
+          return <NotFound label={datasetId} />;
+        }
+
+        if (!question) {
+          return <NotFound label={questionId} />;
```

Comment:
These really need some indication as to what the label is (benchmark, question, ...), but it's fine to leave it for now.